src/persist/src/operators/source.rs

```diff
@@ -56,6 +56,16 @@ where
             // We should probably allow the listen to deregister itself.
             let _ = listen_tx.send(e);
         });
+
+        // We intentionally register the listener before we take the snapshot so
+        // that we know there will be no missing data between the data returned
+        // by the listener and the data contained in the snapshot. However, this
+        // means that either the records received by the listener or the seal
+        // notifications received by the listener could be duplicates of the data
+        // contained in the snapshot and so we have to check every notification to
+        // ensure it is in advance of the snapshot's frontier.
+        // TODO: if we had a way to express "listen as of a time" or "listen after
```

Comment:
I think this TODO is misleading because this isn't possible (or I don't see how it ever will be). It should instead be flipped that we should be able to get information back from the listener about the ts/seqno that it starts at and be able to pass that info into the call to get the snapshot

also nit: empty line before the TODO pls (paragraph break)